server/shards_director.go

```diff
@@ -138,14 +135,18 @@ func (s *shardsDirector) GetOrCreateLeader(namespace string, shardId int64) (Lea
 			return nil, err
 		}
 
-		delete(s.followers, shardId)
+		if _, ok := s.followers[shardId]; ok {
+			s.followersCounter.Dec()
```

Comment:
Is it better call this line behind delete?

Reply:
I think it should be ok either way. We already checked the key is in the map, and the delete will not return error either.